packages/runtime/src/useSSRComputation_Server.tsx

```diff
@@ -12,7 +12,7 @@ export default function useSSRComputation_Server(fn: (...dependencies: any[]) =>
     const cacheKey = calculateCacheKey(relativePathToCwd, dependencies);
     // check if result is a promise
     if (result && typeof result.then === 'function') {
-      cache[cacheKey] = result.then(asyncResult => cache[cacheKey] = asyncResult);
+      throw new Error('useSSRComputation does not support async functions on the server side.');
```

Comment:
This widespread practice of throwing exceptions on invariant violations is inapplicable to Popmenu.

* Popmenu is a CMS, and we don't have 100% code and feature coverage.
* There are so many edge cases, so invariant violations will inevitably spread to production (we deploy every day).
* When this exception is thrown, we will have millions of users waiting for a fix losing money and reputation.

So, please be careful with throwing exceptions unless we are absolutely sure it will never be hit (in this case, we aren't).

Reply:
Saying that we still need to get notified when the hook is misused. Returning a promise server-side is definitely a performance issue, so we want to know about that.

Instead of throwing an exception, please introduce a custom error handling function, so we can call `window.Sentry.captureMessage` Popmenu-side.

I see it as:
```
export function setErrorHandler(errorHandler:(message:string) => void):void {
/**/
}
```
If setErrorHandler isn't called, ignore all invariant violations.

Reply:
Done